measurements/api/measurements.py

```diff
@@ -420,171 +374,228 @@ def list_measurements(
-
-    if confirmed is not None:
-        q = q.filter(c_confirmed == confirmed)
-    if anomaly is not None:
-        q = q.filter(c_anomaly == anomaly)
-    if failure is True:
-        q = q.filter(
-            or_(
-                Measurement.exc != None,
-                Measurement.residual_no != None,
-                c_msm_failure == True,
-            )
-        )
-    if failure is False:
-        q = q.filter(
-            and_(
-                Measurement.exc == None,
-                Measurement.residual_no == None,
-                c_msm_failure == False,
-            )
-        )
-
-    ## Create a query for fastpath
-
+    ## Create fastpath columns for query
     fpcols = [
-        func.coalesce(0).label("m_input_no"),
+        # func.coalesce(0).label("m_input_no"),
         # We use test_start_time here as the batch pipeline has many NULL measurement_start_times
-        Fastpath.measurement_start_time.label("test_start_time"),
-        Fastpath.measurement_start_time.label("measurement_start_time"),
-        func.concat(FASTPATH_MSM_ID_PREFIX, Fastpath.tid).label("measurement_id"),
-        func.coalesce(0).label("m_report_no"),
+        literal_column("measurement_start_time").label("test_start_time"),
+        literal_column("measurement_start_time").label("measurement_start_time"),
+        func.concat(FASTPATH_MSM_ID_PREFIX, sql.text("tid")).label("measurement_id"),
         func.coalesce(
-            (cast(cast(Fastpath.scores["blocking_general"], String), Float) > 0.5)
+            (cast(cast(sql.text("scores->>'blocking_general'"), String), Float) > 0.5)
         ).label("anomaly"),
         func.coalesce(false()).label("confirmed"),
-        func.coalesce(false()).label("msm_failure"),
-        cast(Fastpath.scores.label("scores"), String),
-        func.coalesce([0]).label("exc"),
-        func.coalesce(0).label("residual_no"),
-        Fastpath.report_id.label("report_id"),
-        Fastpath.probe_cc.label("probe_cc"),
-        Fastpath.probe_asn.label("probe_asn"),
-        Fastpath.test_name.label("test_name"),
-        func.coalesce(0).label("report_no"),
+        cast(sql.text("scores"), String).label("scores"),
+        literal_column("report_id"),
+        literal_column("probe_cc"),
+        literal_column("probe_asn"),
+        literal_column("test_name"),
+        literal_column("fastpath.input").label("input"),
     ]
-    if cte is None:
-        fpcols.append(Fastpath.input.label("input"))
-        assert len(fpcols) == len(cols)
-    else:
-        fpcols.append(Fastpath.input.label("input_cte_input"))
-        fpcols.append(func.coalesce(0).label("input_cte_input_no"))
-        assert len(fpcols) == len(cols) + 1
 
-    fpq = current_app.db_session.query(*fpcols)
+    mrwhere = []
+    fpwhere = []
+    query_params = {}
+
+    # Populate WHERE clauses and query_params dict
+
+    if failure is True:
+        # residual_no is never NULL, msm_failure is always NULL
```

Comment:
Hum. The fact that `msm_failure` is always `NULL` sounds like a bug.

I believe that `exc` is used to track exceptions in centrifugation of the measurement file, so as the logic for processing measurements gets better, this should in theory be cleaned up.

In light of the fact that the `failure` value is being set to the `msm_failure` column (https://github.com/ooni/api/blob/master/measurements/api/measurements.py#L426), maybe we should keep the filter as being applied to the msm_failure and document the fact that it's always NULL as a bug in the pipeline?

Reply:
@hellais my suggestion is to ignore residual_no and msm_failure for the moment so that the API queries run faster and the code is leaner. If/when we fix the colums we can add the check back in the API.

Reply:
Ok fair enough.